### PR TITLE
VDB-696 Update transformer execute

### PR DESCRIPTION
**Pull request by elizabethengelman**

### CODE REVIEW

File: libraries/shared/factories/event/repository.go [hunk 1]
```diff
@@ -24,7 +24,6 @@ import (
 type Repository interface {
 	Create(headerID int64, models []interface{}) error
 	MarkHeaderChecked(headerID int64) error
-	MissingHeaders(startingBlockNumber, endingBlockNumber int64) ([]core.Header, error)
```
Comment (elizabethengelman) [inline libraries/shared/factories/event/repository.go]: This transformer specific `MissingHeaders` method was never being used. Instead [this](https://github.com/vulcanize/vulcanizedb/blob/staging/libraries/shared/repository/repository.go#L47) generic `MissingHeaders` method is being called the [watcher](https://github.com/vulcanize/vulcanizedb/blob/staging/libraries/shared/watcher/event_watcher.go#L101) and then the headers are passed into `Transformer.Execute`.

File: libraries/shared/factories/event/transformer.go [hunk 2]
```diff
@@ -37,7 +36,7 @@ func (transformer Transformer) NewTransformer(db *postgres.DB) transformer.Event
 	return transformer
 }
 
-func (transformer Transformer) Execute(logs []types.Log, header core.Header, recheckHeaders constants.TransformerExecution) error {
+func (transformer Transformer) Execute(logs []types.Log, header core.Header) error {
```
Comment (elizabethengelman) [inline libraries/shared/factories/event/transformer.go]: This `recheckHeaders` argument is not being used in this `Execute` method.

File: libraries/shared/repository/repository.go [hunk 3]
```diff
@@ -69,31 +69,6 @@ func MissingHeaders(startingBlockNumber, endingBlockNumber int64, db *postgres.D
 	return result, err
 }
 
-func RecheckHeaders(startingBlockNumber, endingBlockNumber int64, db *postgres.DB, checkedHeadersColumn string) ([]core.Header, error) {
```
Comment (elizabethengelman) [inline libraries/shared/repository/repository.go]: Looks like this method wasn't being called any longer. 

Also, something else I hadn't realized previously was that in order to allow for rechecking headers, we need to pass in a CLI argument `--recheck-headers` when running `composeAndExecute`. 🌠

File: libraries/shared/repository/repository.go [hunk 4]
```diff
@@ -121,37 +96,47 @@ func GetCheckedColumnNames(db *postgres.DB) ([]string, error) {
...
-// Builds a SQL string that checks if any column value is 0, given the column names.
+// Builds a SQL string that checks if any column should be checked/rechecked.
 // Defaults to FALSE when no columns are provided.
 // Ex: ["columnA", "columnB"] => "NOT (columnA!=0 AND columnB!=0)"
 //     [] => "FALSE"
-func CreateNotCheckedSQL(boolColumns []string, recheckHeaders constants.TransformerExecution) string {
-
-	var result bytes.Buffer
-
+func CreateHeaderCheckedPredicateSQL(boolColumns []string, recheckHeaders constants.TransformerExecution) string {
```
Comment (elizabethengelman) [inline libraries/shared/repository/repository.go]: I refactored this method into two smaller ones & changed the name as I was trying to understand how it was working. If folks don't find this refactoring helpful, I'm happy to undo it.

File: libraries/shared/repository/repository_test.go [hunk 5]
```diff
@@ -153,10 +149,11 @@ var _ = Describe("Repository", func() {
 		})
 
 		AfterEach(func() {
-			test_config.CleanCheckedHeadersTable(db, getExpectedColumnNames())
+			//test_config.CleanCheckedHeadersTable(db, getExpectedColumnNames())
```
Comment (rmulhol) [inline libraries/shared/repository/repository_test.go]: Do we want to remove this `AfterEach` entirely?

File: libraries/shared/repository/repository_test.go [hunk 6]
```diff
@@ -285,33 +246,51 @@ var _ = Describe("Repository", func() {
...
 	})
 
-	Describe("CreateNotCheckedSQL", func() {
-		It("generates a correct SQL string for one column", func() {
-			columns := []string{"columnA"}
-			expected := "NOT (columnA!=0)"
-			actual := shared.CreateNotCheckedSQL(columns, constants.HeaderMissing)
-			Expect(actual).To(Equal(expected))
-		})
+	Describe("CreateCustomColumnNamesSQL", func() {
```
Comment (rmulhol) [inline libraries/shared/repository/repository_test.go]: Should this be `"CreateHeaderCheckedPredicateSQL"`?

File: libraries/shared/repository/repository_test.go [hunk 7]
```diff
@@ -285,33 +246,51 @@ var _ = Describe("Repository", func() {
...
+				Expect(actual).To(Equal(expected))
+			})
 		})
 
-		It("generates a correct SQL string for rechecking headers", func() {
-			columns := []string{"columnA", "columnB"}
-			expected := fmt.Sprintf("NOT (columnA>=%s AND columnB>=%s)", constants.RecheckHeaderCap, constants.RecheckHeaderCap)
-			actual := shared.CreateNotCheckedSQL(columns, constants.HeaderRecheck)
-			Expect(actual).To(Equal(expected))
+		Describe("RecheckHeadersCustomColumnNames", func() {
```
Comment (rmulhol) [inline libraries/shared/repository/repository_test.go]: maybe rephrase to clarify how it's different from the above `Describe`

File: libraries/shared/repository/repository.go (hunk 4 above)
Comment (i-norden) [reply libraries/shared/repository/repository.go]: I think it makes more sense this way 💯

File: libraries/shared/repository/repository_test.go (hunk 5 above)
Comment (elizabethengelman) [reply libraries/shared/repository/repository_test.go]: oh nice catch!  i think I commented this out so I could get some data in my test db - I think it should still hang around.
